### PR TITLE
[ML] Platform independent dependency report script

### PR DESCRIPTION
**Pull request by edsavage**
Replace dependency_report.sh bash script with cmake script equivalent

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -428,7 +428,7 @@ task buildDependencyReport(type: Exec) {
   outputs.file("${buildDir}/distributions/dependencies-${version}.csv")
   environment makeEnvironment
   commandLine shell
-  args '-c', "source ./set_env.sh && 3rd_party/dependency_report.sh --csv \"${outputs.files.singleFile}\""
+  args shell_flag, "${set_env} && ${cmake} -D OUTPUT_FILE=\"${outputs.files.singleFile}\" -P 3rd_party/dependency_report.cmake"
```
Comment (edsavage) [inline build.gradle:431]: TODO: change variables here to camel case once #2347 has been merged to main

Comment (droberts195) [reply build.gradle:431]: You should be able to merge main into this PR branch now to get the updates

File: 3rd_party/dependency_report.cmake [hunk 2]
```diff
@@ -15,46 +15,45 @@
 # that is required for the stack-wide list.
 #
 # Usage:
-# dependency_report.sh --csv <output_file>
+# cmake -D OUTPUT_FILE=<output_file> -P dependency_report.cmake
```
Comment (droberts195) [inline 3rd_party/dependency_report.cmake:18]: Please keep the existing `dependency_report.sh` in parallel to this new script until we can move release manager over to the new script.

**Comment by droberts195** [comment]
I think you should merge latest main into the PR branch because then the PR will show just the files changed in this PR.  At the moment it's showing changes from other PRs that weren't merge clashes with main, so are completely unrelated.

**Pull request closed without merging**